site/static/js/plugins.js

```diff
@@ -152,6 +152,38 @@ function processAsciinemaTags(content) {
   );
 }
 
+function addCodeCopyButtons() {
+  const preBlocks = Array.from(document.getElementsByTagName("pre")).filter(
+    (el) => el.classList.contains("language-shell")
+  );
+  const makeButton = () => {
+    const copyButton = document.createElement("button");
+    const buttonClassName = "copy-button";
+    copyButton.classList.add(buttonClassName);
+    copyButton.title = "Copy to clipboard";
+
+    const copyIcon = document.createElement("span");
+    copyIcon.innerText = "copy";
+    copyIcon.classList.add("material-icons-outlined");
+    copyButton.appendChild(copyIcon);
+
+    copyButton.addEventListener("click", (el) =>
+      navigator.clipboard.writeText([
+        el
+          .composedPath()
+          .find((el) => el.classList.contains(buttonClassName))
+          .previousElementSibling.textContent.split("\n")
+          .map((s) => s.trimLeft())
+          .filter((s) => s.startsWith("$") || s.startsWith("|"))
```

Comment:
- Multi-line should be based on line continuation character not pipes (make sure to test this).
- If a shell block doesn't have '$', don't display the icon. This is defense in-depth even if we have a linter (which we don't today).
- This also needs to be done for catalog site after auditing it for correct usage of shell blocks
- We should enable docs linter ASAP

Reply:
- Automated test for this.
- Automated test for this.
- We should be able to refer to plugins in production from the catalog site. Happy to test it after this PR is merged.
- I agree; it looks like an m4 task https://github.com/GoogleContainerTools/kpt/issues/1652 Since it'll take some time, it should be bumped up to m3 if we want to prioritize